dome9/resource_dome9_admission_control_policy.go

```diff
@@ -0,0 +1,121 @@
+package dome9
+
+import (
+	"github.com/dome9/dome9-sdk-go/dome9/client"
+	"github.com/dome9/dome9-sdk-go/services/admissioncontrol/admission_policy"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"log"
+)
+
+func resourceAdmissionPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAdmissionControlPolicyCreate,
+		Read:   resourceAdmissionControlPolicyRead,
+		Update: resourceAdmissionControlPolicyUpdate,
+		Delete: resourceAdmissionControlPolicyDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"target_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"target_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"Environment", "OrganizationalUnit"}, false),
+			},
+			"ruleset_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"notification_ids": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"action": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"Prevention", "Detection"}, false),
+			},
+		},
+	}
+}
+
+func resourceAdmissionControlPolicyCreate(d *schema.ResourceData, meta interface{}) error {
+	d9Client := meta.(*Client)
+	req := expandAdmissionControlPolicyRequest(d)
+	log.Printf("[INFO] Creating Admission Control policy request %+v\n", req)
+	resp, _, err := d9Client.admissionControlPolicy.Create(&req)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Created admission control Policy with ID: %v\n", resp.ID)
+	d.SetId(resp.ID)
+
+	return resourceAdmissionControlPolicyRead(d, meta)
+}
+
+func expandAdmissionControlPolicyRequest(d *schema.ResourceData) admission_policy.AdmissionControlPolicyRequest {
+	return admission_policy.AdmissionControlPolicyRequest{
+		TargetId:        d.Get("target_id").(string),
+		RulesetId:       d.Get("ruleset_id").(int),
+		NotificationIds: expandNotificationIDs(d, "notification_ids"),
```

Comment:
this is not good practice to connect between two unrelated file. usually in such case we need some common file for methods that both file will use. @itzikYeret what do you think? should we open a new file under common folder and move this method from compliance policy to there and both will use? or it can stay like this?